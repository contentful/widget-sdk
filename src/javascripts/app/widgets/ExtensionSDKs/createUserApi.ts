import { UserAPI } from '@contentful/app-sdk';
import { SpaceMember } from 'classes/spaceContextTypes';

export const createUserApi = (spaceMember: SpaceMember): UserAPI => {
  return {
    sys: {
      type: 'User',
      id: spaceMember.sys.user.sys.id,
    },
    firstName: spaceMember.sys.user.firstName,
    lastName: spaceMember.sys.user.lastName,
    email: spaceMember.sys.user.email,
    avatarUrl: spaceMember.sys.user.avatarUrl,
    spaceMembership: {
      sys: {
        type: 'SpaceMembership',
        id: spaceMember.sys.id,
      },
      admin: !!spaceMember.admin,
      roles: spaceMember.roles.map((role) => ({
        name: role.name,
        description: role.description ?? '',
      })),
    },
  };
};
