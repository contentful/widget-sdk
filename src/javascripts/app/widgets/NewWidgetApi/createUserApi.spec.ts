import { createUserApi, SpaceMember } from './createUserApi';

describe('createUserApi', () => {
  const spaceMember: SpaceMember = {
    admin: false,
    roles: [{ name: 'dev', description: 'thing' }],
    sys: {
      id: 'member_id',
      user: {
        sys: {
          id: 'user_id',
        },
        firstName: 'bob',
        lastName: 'cratchit',
        email: 'bob@bob.com',
        avatarUrl: 'www.example.com',
      },
    },
  };

  it('formats the spacemember correctly', () => {
    const userApi = createUserApi(spaceMember);
    expect(userApi).toMatchSnapshot();
  });
});
