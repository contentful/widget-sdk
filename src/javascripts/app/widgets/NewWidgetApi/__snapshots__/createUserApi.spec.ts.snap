// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUserApi formats the spacemember correctly 1`] = `
Object {
  "avatarUrl": "www.example.com",
  "email": "bob@bob.com",
  "firstName": "bob",
  "lastName": "cratchit",
  "spaceMembership": Object {
    "admin": false,
    "roles": Array [
      Object {
        "description": "thing",
        "name": "dev",
      },
    ],
    "sys": Object {
      "id": "member_id",
      "type": "SpaceMembership",
    },
  },
  "sys": Object {
    "id": "user_id",
    "type": "User",
  },
}
`;
